R/glue.R

```diff
@@ -65,19 +65,19 @@ glue_data <- function(.x, ..., .sep = "", .envir = parent.frame(),
 
   # Perform all evaluations in a temporary environment
   if (is.null(.x)) {
-    env <- new.env(parent = .envir)
+    parent <- .envir
   } else if (is.environment(.x)) {
-    env <- new.env(parent = .x)
+    parent <- .x
   } else {
-    env <- list2env(.x, parent = .envir)
+    parent <- list2env(.x, parent = .envir)
   }
 
   # Capture unevaluated arguments
   dots <- eval(substitute(alist(...)))
   named <- has_names(dots)
 
   # Evaluate named arguments, add results to environment
-  assign_args(dots[named], env)
+  env <- bind_args(dots[named], parent)
```

Comment:
Name changed to `bind_args()` to reflect that a new environment is created, rather mutating an existing one.